### PR TITLE
rgw: zone service only provides const access to its data

### PR DESCRIPTION
**Pull request by cbodley**

### CODE REVIEW

**Comment by cbodley** [comment]
http://pulpito.ceph.com/cbodley-2018-12-11_14:43:34-rgw-wip-cbodley-testing-distro-basic-smithi/

File: src/rgw/services/svc_zone.cc [hunk 1]
```diff
@@ -1010,19 +1010,12 @@ int RGWSI_Zone::select_bucket_location_by_rule(const string& location_rule, RGWZ
    * checking it for the local zone, because that's where this bucket object is going to
    * reside.
    */
-  map<string, RGWZonePlacementInfo>::iterator piter = get_zone_params().placement_pools.find(location_rule);
-  if (piter == get_zone_params().placement_pools.end()) {
+  auto piter = zone_params->placement_pools.find(location_rule);
+  if (piter == zone_params->placement_pools.end()) {
     /* couldn't find, means we cannot really place data for this bucket in this zone */
-    if (get_zonegroup().equals(zonegroup->get_id())) {
```
Comment (theanalyst) [inline src/rgw/services/svc_zone.cc]: is this condition always true?

Comment (theanalyst) [reply src/rgw/services/svc_zone.cc]: yep looks like it

Comment (cbodley) [reply src/rgw/services/svc_zone.cc]: i think this was intended to compare the running zonegroup against the _bucket's_ zonegroup

**Pull request merged**